### PR TITLE
Support additional GCPManagedControlPlane fields

### LINKED ISSUE #906: Support additional fields for GKE
/kind feature

**Describe the solution you'd like**

CAPG initial support for Managed Cluster (GKE) has added few basic configurations. GCP does support lot more fields for example private cluster, enable workload identity, just to name a few. This is a request to add support for more fields than just basics.

[This spreadsheet](https://docs.google.com/spreadsheets/d/1ETX0JyIL5jiOWPUrxOSOJ8yCzXvSERp2bZ3jPn33Vdo/edit#gid=0) enlists all important GKE  configuration we can support in GCPManaged* kinds.

All additional GKE fields are documented in the spreadsheet linked above. The structure in spreadsheet can be used for defining GCPManaged API specs. Feel free to comment to discuss more about any fields.

**Anything else you would like to add:**

There are other open issues those requested for additional fields too:
- https://github.com/kubernetes-sigs/cluster-api-provider-gcp/issues/903
- https://github.com/kubernetes-sigs/cluster-api-provider-gcp/issues/897

Such requests can be tackled by the complete list (spreadsheet) attached here.

If this sounds good then, I am happy to take this forward and decide sub-tasks for better collaboration.

looping @poornima-rafay @niravparikh05

**Progress**

```[tasklist]
### Tasks - PR1
- [x] GCPManagedMachinePool changes (@akshay196) - https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/944
```

```[tasklist]
### Tasks - PR2
- [ ] GCPManagedControlPlane changes - part 1 (Description, ClusterNetwork, DefaultNodeLocation, DefaultMaxPodsPerNode) - ~`PR#936` by Anniruddha~ (@akshay196) - https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/1186
```

```[tasklist]
### Tasks - PR3
- [ ] GCPManagedControlPlane changes - part 2 (ClusterSecurity, AddonsConfig, LoggingConfig, MonitoringConfig) (@akshay196) - https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/1186
```

```[tasklist]
### Tasks - PR4
- [ ] Update relevant documentation if any
```

### PR DESCRIPTION
**Pull request by akshay196-rafay**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Add following GCPManagedControlPlane fields:

- Description
- Cluster Network
  - PrivateCluster
  - UseIPAliases
  - Pod
  - Service
- Cluster Security
  - WorkloadIdentityConfig
  - AuthenticatorGroupConfig
  - EnableLagacyAuthorization
  - IssueClientCertificate
- AddonsConfig
- LoggingConfig
- MonitoringConfig

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #906 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
TBD
```

### CODE REVIEW

**Comment by richardcase** [comment]
@akshay196-rafay - would you be able to look at the lint issues?

**Comment by richardcase** [comment]
@akshay196-rafay @aniruddha2000 - would you be able to do a `make generate`? Lets try and get this PR merged.

**Comment by FrancoisKlieberOrange** [comment]
Hi @akshay196-rafay @aniruddha2000,

I hope you're doing well.

I just wanted to reach out to see if there's anything I can do to help you with it. If you're stuck on something or could use an extra set of eyes to review or troubleshoot, I'd be more than happy to assist.

Please feel free to let me know how I can support you, whether it's discussing the issue, collaborating on a solution, or just providing some feedback. I'm here to help!

Best regards,
Francois

**Comment by belgaied2** [comment]
@cpanato @richardcase this PR has been dramatically reduced in scope in the hope that it will be merged faster.
The focus is set on GKE Private clusters.

Some effort has been done to have all CI tests pass. Please, review and let me know if I should improve anything.

File: exp/api/v1beta1/gcpmanagedcontrolplane_types.go [hunk 1]
```diff
@@ -27,13 +27,104 @@ const (
...
+	PrivateCluster *PrivateCluster `json:"privateCluster,omitempty"`
+
+	// UseIPAliases is whether alias IPs will be used for pod IPs in the cluster. If false, routes will be used for
+	// pod IPs in the cluster.
+	// +optional
+	UseIPAliases bool `json:"useIPAliases,omitempty"`
+
+	// Pod defines the range of CIDRBlock list from where it gets the IP address.
+	// +optional
+	Pod *ClusterNetworkPod `json:"pod,omitempty"`
```
Comment (dims) [inline exp/api/v1beta1/gcpmanagedcontrolplane_types.go:89]: We don't use this field in this PR ... do we?

File: exp/api/v1beta1/gcpmanagedcontrolplane_types.go [hunk 2]
```diff
@@ -27,13 +27,104 @@ const (
...
+
+	// Service defines the range of CIDRBlock list from where it gets the IP address.
+	// +optional
+	Service *ClusterNetworkService `json:"service,omitempty"`
```
Comment (dims) [inline exp/api/v1beta1/gcpmanagedcontrolplane_types.go:93]: Same question as above

File: exp/api/v1beta1/gcpmanagedcontrolplane_types.go (hunk 1 above)
Comment (afarbos) [reply exp/api/v1beta1/gcpmanagedcontrolplane_types.go:89]: Exactly, I set those expecting them to work. those don't do anything.

**Pull request merged**